scripts/objectstore/copy_files_to_irods.py

```diff
@@ -140,9 +143,31 @@ def copy_files_to_irods(start_dataset_id, end_dataset_id, object_store_info_file
             session.collections.create(irods_file_collection_path)
 
             # Add disk file to collection
-            options = {kw.DEST_RESC_NAME_KW: 'demoResc'}
+            options = {kw.DEST_RESC_NAME_KW: 'demoResc', kw.REG_CHKSUM_KW : ''}
             session.data_objects.put(disk_file_path, irods_file_path, **options)
 
+            # Calculate disk file checksum before uploading it to irods
+            # After disk file is uploaded to irods, we get the file checksum from irods and compare it with the calculated disk file checksum
+            # Note that disk file checksum is ASCII, whereas irods file checksum is Unicode
+            disk_file_checksum = get_file_checksum(disk_file_path)
```

Comment:
@kxk302 we once discussed to calculate for all files a checksum in Galaxy and store this as part of every dataset metadata in Galaxy. This will enable many interesting use-cases maybe its really time to get this done, so you do not need to calculate it here.

Reply:
@bgruening the checksum calculation is automatically done bu irods (You just need to set a configuration flag). whatI'm doing here is calculating the checksum outside irods, so I can compare to irods checksum, as test for upload to irods being valid.